libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -343,34 +347,39 @@ proc handleGraft(g: GossipSub,
                  peer: PubSubPeer,
                  grafts: seq[ControlGraft],
                  respControl: var ControlMessage) =
+  let peerId = peer.id
   for graft in grafts:
-    trace "processing graft message", peer = peer.id,
-                                      topicID = graft.topicID
-
-    if graft.topicID in g.topics:
-      if g.mesh.len < GossipSubD:
-        g.mesh[graft.topicID].incl(peer.id)
+    let topic = graft.topicID
+    trace "processing graft message", topic, peerId
+
+      # If they send us a graft before they send us a subscribe, what should
```

Comment:
I would say that there is no harm in adding them to gossipsub, if anything it will keep the tables in sync.